### ISSUE TITLE
General Discussion - 1.0 Release Candidate

### ISSUE THREAD

**Issue by NLKNguyen** [NLKNguyen/papercolor-theme#71]
This thread is for any discussion regarding 1.0 release candidate

See [ROADMAP](https://github.com/NLKNguyen/papercolor-theme/blob/master/ROADMAP.md#10-release-candidate) for information

**Comment by pickfire** [comment]
Oh, I didn't know https://github.com/NLKNguyen/papercolor-16 happened to be here. I hope that it can be linked to the README.

**Comment by NLKNguyen** [comment]
Some updates on [develop_1.0](https://github.com/NLKNguyen/papercolor-theme/tree/develop_1.0) branch: 
The goal of the current refactoring process is to allow color theme to be designed in this manner:
```VimL
let s:themes['papercolor'].dark = {
      \     'palette' : {
      \       'color00' : ['#262626', '234'],
      \       'color01' : ['#df0000', '100'], 
      \       'color02' : ['#5faf00', '70'],
      \       'color03' : ['#dfaf5f', '179'], 
      \       'color04' : ['#303030', '235'],
      \       'color05' : ['#578787', '29'], 
      \       'color06' : ['#df875f', '173'], 
      \       'color07' : ['#d0d0d0', '251'], 
      \       'color08' : ['#8a8a8a', '244'], 
      \       'color09' : ['#5faf5f', '71'],
      \       'color10' : ['#afdf00', '148'],
      \       'color11' : ['#af87df', '140'], 
      \       'color12' : ['#444444', '237'], 
      \       'color13' : ['#ff5faf', '205'],
      \       'color14' : ['#00afaf', '37'], 
      \       'color15' : ['#5fafdf', '74'],
" other optional colors ...
```
The color guideline is updated here http://codepen.io/NLKNguyen/full/NrRpej/ It has some changes since the last time.

The first 15 color## are required and can be mapped  one-to-one directly to the color palette of the terminal. For instance, https://github.com/NLKNguyen/papercolor-16/tree/master/windows-command-prompt 
Default: 
![cmd-00-papercolor-default](https://cloud.githubusercontent.com/assets/4667129/16972261/d6ee2d58-4dde-11e6-8cb9-f8d9e4635b23.PNG)
After installing PaperColor-Dark by double clicking on the .reg file:
![cmd-01-papercolor-dark](https://cloud.githubusercontent.com/assets/4667129/16972263/d914f940-4dde-11e6-93d2-b65c00d16fbf.PNG)
In order to not having to restart Windows to see the changes after running .reg file, you might have to right click on desktop and click "open command window here". By some reason, that session is up-to-date.

**Comment by NLKNguyen** [comment]
With the current color guideline, some cli programs can produce the natural color, for example, red and green for deleting and adding in Git
![cmd-02-papercolor-gitdiff](https://cloud.githubusercontent.com/assets/4667129/17008339/3dfd5b38-4ea5-11e6-9dc6-f87c5bbc3994.PNG)

**Comment by NLKNguyen** [comment]
To test 16-color in Vim using a 256-color terminal, you need to `set t_Co=8`

**Comment by pickfire** [comment]
Yes, it works and follows most of the colors in 8-color terminal except blue but those colors from 9 - 16 is very different from the original color expecially the last one (white -> blue).

**Comment by NLKNguyen** [comment]
There is no way around. All colors are to be alternated, which is how we can have a different palette. I imagine other palettes like PaperColor light, Solarized, Gruvbox, etc can be added as well, and it's not possible to not alternate the default colors. The red and green, however, have more special cognition meaning associated with them, so they are better to be remained.

I intend to have the 16-color theme to looks good even with the default terminal color palette. I expect it to look similar across different terminals, but I haven't tested anywhere except Windows CMD. If you can test how it looks on some Linux terminals (w/ default palette) and upload the screenshots, that'd be very helpful to me.

By 9-16, you meant 00 to 15 right? In the PaperColor.vim I have color 16 and 17, but they are optional as well as the rest. All optional colors will have sensible default, being drawn from the first 15.

**Comment by pickfire** [comment]
Sure, why not?

@NLKNguyen Here's the default color, which is high in contrast but now it suits my eyes more.

![2016-07-22-004713_564x340_scrot](https://cloud.githubusercontent.com/assets/4687791/17031318/05a768e4-4fa6-11e6-9e6b-16039dcec34e.png)
Light theme: assembly
![2016-07-22-005807_564x340_scrot](https://cloud.githubusercontent.com/assets/4687791/17031591/77bf71c8-4fa7-11e6-9cba-55e2a5a2c8aa.png)
Dark theme: assembly
![2016-07-22-005815_564x340_scrot](https://cloud.githubusercontent.com/assets/4687791/17031590/77bc2252-4fa7-11e6-8580-e20d96d1f5f7.png)

**Comment by NLKNguyen** [comment]
You need to set t_Co=8 to force vim to use native  terminal 16 colors. The above used 256.

**Comment by pickfire** [comment]
@NLKNguyen t_Co=8 doesn't seem to have an effect in neovim. And I haven't set my terminal colors to papercolor yet.

**Comment by ShengYun** [comment]
@NLKNguyen Is it possible to change the status bar color in the light theme? now it's a bit hard to distinguish the horizontal split. See screenshot below

![papercolorlightsplit](https://cloud.githubusercontent.com/assets/1317203/18496462/12adcdbc-79f4-11e6-97f9-24664933b21e.png)

**Comment by NLKNguyen** [comment]
I thought it would be more visible than that. Are you sure it's not your Airline/Lightline theme?
This is what I have with plain Vim statusline. I could still darken it a bit more if necessary.
![capture](https://cloud.githubusercontent.com/assets/4667129/18496716/fc06fb72-79dc-11e6-9a7a-d36a3627b63e.PNG)

**Comment by pickfire** [comment]
@NLKNguyen Ah, what about using orange for it, looks nice.

**Comment by ShengYun** [comment]
@NLKNguyen Am I supposed to install a separate airline theme? I'm currently using ` vim-airline/vim-airline-themes` and did not set anything about the airline theme in my `vimrc`

**Comment by NLKNguyen** [comment]
@ShengYun Oh I see. You're using the Airline theme provided in this repo. There will be a revision on Airline theme after version 1.0 release, and I will use a separate repo for Airline. For now I'll be working on the main theme only.

@pickfire I will consider that :)

**Comment by ShengYun** [comment]
@NLKNguyen Oh so what you are saying is after the 1.0 release I should install another repo and use that theme? but keep using the built in airline theme for now?

**Comment by NLKNguyen** [comment]
Yeah, the Airline theme is unmaintained now and will be replaced by a different, separate project. It's gonna be a long time until then so keep using the current Airline or alternative Lightline somewhere if you like.
